content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+
+The Prisma schema file currently looks as follows:
+
+```prisma file=prisma/schema.prisma
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+<Tip>
+
+If you're using VS Code, be sure to install the [Prisma VS Code extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) for syntax highlighting, formatting, auto-completion and a lot more cool features.
+
+</Tip>
+
+### 2.2. Connect your database
+
+You can configure your [database connection URL](/reference/database-reference/connection-urls#mongodb) in the `.env` file.
+
+The format of the connection URL Mongoose uses is similar to the one Prisma will use.
+
+```bash file=.env
+mongodb://alice:myPassword43@localhost:27017/blog-mongoose
+```
+
+Refer to the [MongoDB connection URL specification](https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-options) for further details.
+
+### 2.3. Run Prisma's introspection
+
+With your connection URL in place, you can [introspect](/concepts/components/introspection) your database to generate your Prisma models:
+
+> **Note**: MongoDB is a _schemaless_ database. To incrementally adopt Prisma in your project, ensure your database is populated with sample data. Prisma introspects a MongoDB schema by sampling data stored and inferring the schema from the data in the database
+
+```terminal copy
+npx prisma db pull
+```
+
+This creates the following Prisma models:
+
+```prisma file=prisma/schema.prisma
+type UsersProfile {
+  bio String
+}
+
+model categories {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  v    Int    @map("__v")
+  name String
+}
+
+model posts {
+  id         String   @id @default(auto()) @map("_id") @db.ObjectId
+  v          Int      @map("__v")
+  author     String   @db.ObjectId
+  categories String[] @db.ObjectId
+  content    String
+  published  Boolean
+  title      String
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+}
+```
+
+The generated Prisma models represent the collections and are the foundation of your programmatic Prisma Client API which allows you to send queries to your database.
+
+### 2.4. Adjust the relationships
+
+MongoDB doesn't support relations between different collections. However, you can create references from one document to another using the `ObjectId` schema field or from one document to many using an array of `ObjectIds`. The reference will store id(s) of the related document(s). You can use the `populate()` method Mongoose provides to populate the reference with the data of the related document.
```

Comment:
> using the `ObjectId` schema field

I think this is not super precise. You can use a field of that type that will contain value that refer to the values of the _id (or other) field in that collection.